pkg/client/cache/reflector_test.go

```diff
@@ -375,7 +376,24 @@ func TestReflectorResync(t *testing.T) {
 	}
 	resyncPeriod := 1 * time.Millisecond
 	r := NewReflector(lw, &api.Pod{}, s, resyncPeriod)
-	r.ListAndWatch(stopCh)
+	clock := util.NewFakeClock(time.Now())
+	r.clock = clock
+
+	go r.ListAndWatch(stopCh)
+
+	// advance the clock until the reflector stops
+Loop:
```

Comment:
I do not think this solves the problem. If resyncfunc hits line 362 while this for loop is at line 393, the timer will still be increased at line 387. Then listAndWatch might call resyncfunc again.